src/index.js

```diff
@@ -18,40 +19,106 @@ function runPackageScripts({scriptConfig, scripts, args, options = {}}, callback
     scripts = ['default']
   }
   const scriptNames = arrify(scripts)
-  const method = options.parallel ? 'map' : 'mapSeries'
-  async[method](scriptNames, (scriptName, cb) => {
-    const child = runPackageScript({scriptConfig, options, scriptName, args})
-    if (child.on) {
-      child.on('exit', exitCode => cb(null, exitCode))
-    } else {
-      cb(child)
+  if (options.parallel) {
```

Comment:
Maybe [`zip`](https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/operators/zip.md) is what you're looking for (re RxJS)? Merge multiple Observables/Promises, subscribe to the result and have ha single handling of their results ...